freqtrade/exchange/exchange.py

```diff
@@ -1983,6 +1982,126 @@ def get_funding_fees(
         else:
             return 0.0
 
+    @retrier
+    def get_liquidation_price(
+        self,
+        pair: str,
+        # Dry-run
+        open_rate: Optional[float] = None,   # Entry price of position
+        is_short: Optional[bool] = None,
+        position: Optional[float] = None,  # Absolute value of position size
+        wallet_balance: Optional[float] = None,  # Or margin balance
+        mm_ex_1: float = 0.0,  # (Binance) Cross only
+        upnl_ex_1: float = 0.0,  # (Binance) Cross only
+    ):
+        """
+        Set's the margin mode on the exchange to cross or isolated for a specific pair
+        :param pair: base/quote currency pair (e.g. "ADA/USDT")
+        """
+        if self.trading_mode == TradingMode.SPOT:
+            return None
+        elif (self.collateral is None):
+            raise OperationalException(f'{self.name}.collateral must be set for liquidation_price')
+        elif (self.trading_mode != TradingMode.FUTURES and self.collateral != Collateral.ISOLATED):
+            raise OperationalException(
+                f"{self.name} does not support {self.collateral.value} {self.trading_mode.value}")
+
+        if self._config['dry_run'] or not self.exchange_has("fetchPositions"):
+            if (
+                open_rate is None or
+                is_short is None or
+                position is None or
+                wallet_balance is None
```

Comment:
I'd make all of these parameters required (basically removing the Optional - and the `is None` checks that go along with that.

`get_liquidation_price` is only called from one place - where all parameters are "force-provided" (they come in as non-optional) - so having all this "optional" handling only blows the whole method up.
the only other place i can imagine this method to be called is from backtesting, and there we MUST provide these parameters anyway - as that'll have to use dry-run calculations.